enhancements/workload-partitioning/rationalizing-configuration.md

```diff
@@ -0,0 +1,460 @@
+14. The admission plugin uses the workload types on the new enablement
+    API to decide when to mutate pods.
+15. CRI-O sees pods with workload annotations and uses the resource
+    request to set cpushares and cpuset.
+16. On day 2 PAO, is installed and takes ownership of the
+    PerformanceProfile CRs.
+
+#### Day 2: Modify Reserved cpuset for a PerformanceProfile
+
+In a cluster with partitioning enabled and fully configured, modifying
+the reserved cpuset for a PerformanceProfile is safe because the node
+is rebooted when the new MachineConfig is applied.
+
+1. The user updates the cpuset in the PerformanceProfile(s).
+2. PAO generates new MachineConfigs, adding the cpuset information
+   from the PerformanceProfiles to the other partitioning info from
+   the new enablement API.
+3. The MCO rolls out new MachineConfigs, rebooting nodes as it goes.
+4. The cluster has complete partitioning configuration for management
+   workloads and the management workloads are partitioned due to the
+   nodes rebooting.
+5. Kubelet starts and sees the config file enabling partitioning. It
+   advertises the workload resource type on the Node and mutates
+   static pods with partitioning annotations.
+6. Admission plugin uses the workload types on the workloads CR to
+   decide when to mutate pods.
+7. CRI-O sees pods with workload annotations and uses the resource
+   request to set cpushares and cpuset.
+
+#### Day 2: Add a New Node to an Existing MachineConfigPool
+
+In a cluster with partitioning enabled and fully configured, adding a
+new node to an existing MachineConfigPool is safe because the
+MachineConfig will set up kubelet and CRI-O on the host correctly.
+
+1. The MCO applies the MachineConfigs to the node.
+2. Kubelet starts and sees the config file enabling partitioning. It
+   advertises the workload resource type on the Node and mutates
+   static pods with partitioning annotations.
+3. CRI-O sees pods with workload annotations and uses the resource
+   request to set cpushares and cpuset.
+
+#### Day 2: Add a New MachineConfigPool
+
+Adding a new MachineConfigPool is safe only if the MachineConfigPool
+has a PerformanceProfile associated and the PAO is installed.
+
+1. PAO reads PerformanceProfile to generate new MachineConfig for the
+   pool with CRI-O and `kubelet` settings.
+2. MachineConfigs are applied to appropriate nodes, rebooting them in
+   the process.
+3. Kubelet starts and sees the config file enabling partitioning. It
+   advertises the workload resource type on the Node and mutates
+   static pods with partitioning annotations.
+4. CRI-O sees pods with workload annotations and uses the resource
+   request to set cpushares and cpuset.
+
+If the MachineConfigPool does not match a PerformanceProfile, there
+will be no cpuset information and the PAO will generate MachineConfigs
+with partitioning enabled but not tied to a cpuset. This is somewhat
+safe, but may lead to unexpected behavior.
```

Comment:
The mutated pods would float across the full cpuset as they would without the feature enabled, however the cpu shares would not get deducted from alloctable cpus potentially leading to over commit scenarios.

Reply:
Thanks, I've included this info in this section.